File: servers/apitracker/models/req.js
const mongoose = require('mongoose');




// method = StringField(required=True)
// endpoint = StringField(required=True)
// ip = StringField()
// accessed_on = DateTimeField(default=datetime.now)



const reqSchema = mongoose.Schema({
    method: {
        type: String,
        required: true
    },
    endpoint: {
        type: String,
        required: true
    },
    ip: {
        type: String
    },
    accessed_on: {
        type: Date,
        default: Date.now
    }
});

const req = mongoose.model('req', reqSchema);


module.exports = req;

File: servers/api/models/req.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from mongoengine import *
from datetime import datetime


class Reqs(Document):
    ''' Creates request instance '''

    method = StringField(required=True)
    endpoint = StringField(required=True)
    ip = StringField()
    accessed_on = DateTimeField(default=datetime.now)


    def to_json(self):
        ''' returns document as json '''
        return {
            "method": str(self.form),
            "endpoint": str(self.endpoint),
            "called_on": str(self.called_on)
        }
